src/vm/methodtablebuilder.cpp

```diff
@@ -5344,6 +5377,12 @@ MethodTableBuilder::PlaceVirtualMethods()
                                 it.Token());
                     }
                 }
+
+                // mazong1123: a valid overrided method.
+                if (!*hasOverridedEqualsMethod)
+                {
```

Comment:
If I understand correctly, try to find an overrided `Equals` when somebody calls Equals in a ValueType will need to DFS or BFS all the available fields of the types in worse case. If the hierarchy is deep it's very inefficient IMO, for instance:

```
    struct MyStruct2
    {
        public NeverEquals _x;
        public NeverEquals _y;
    }

    struct MyStruct3
    {
        public MyStruct2 _x;
        public MyStruct2 _y;
    }

    struct NeverEquals
    {
        public double _a;

        public override bool Equals(object obj)
        {
            return false;
        }

        public override int GetHashCode()
        {
            return 0;
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            NeverEquals i = new NeverEquals();
            NeverEquals j = new NeverEquals();

            Console.WriteLine(i.Equals(j));
        }
    }
```

We need to drill down to `NeverEquals` to find an overrided `Equals`. That makes the "fast equal" no more fast... 

I think we may need some trade-off. For example, something like "if you override any method in a ValueType, `CanCompareBits` will return false`. Because we can easily record whether the user override any method in the hierarchy, we can simply check it in `CanCompareBits`.

Similar to floating point number check. We can easily record whether the user use any floating point number field in the hierarchy, but it's hard to check if any floating point number filed stores `0.0` or `NaN` at runtime (Need a DFS or BFS).

@jkotas please let me know if I missed something. Thanks.